src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -146,8 +148,16 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
         if (!(participation instanceof ProgrammingExerciseParticipation programmingExerciseParticipation)) {
             return notFound();
         }
-        if (!programmingExerciseParticipationService.canAccessParticipation(programmingExerciseParticipation)
-                || (submissionType.equals(SubmissionType.INSTRUCTOR) && !authCheckService.isAtLeastInstructorForExercise(participation.getExercise()))) {
+
+        if (!programmingExerciseParticipationService.canAccessParticipation(programmingExerciseParticipation)) {
+            return forbidden();
+        }
+
+        // The editor is allowed to trigger an instructor type build for template and solution participations,
```

Comment:
```suggestion
        // The editor is allowed to trigger an instructor build for template and solution participations,
```
A matter of taste, can be ignored.